prow/cmd/deck/main.go

```diff
@@ -1274,8 +1286,24 @@ func canTriggerJob(user string, pj prowapi.ProwJob, cfg *prowapi.RerunAuthConfig
 	}
 	if cli == nil {
 		logrus.Warning("No GitHub token was provided, so we cannot retrieve GitHub teams")
+		return false, nil
+	}
+
+	// If the job is a presubmit and has an associated PR, do the same checks as for /test
+	if pj.Spec.Type != prowapi.PresubmitJob || pj.Spec.Refs == nil || len(pj.Spec.Refs.Pulls) == 0 {
+		return false, nil
+	}
+	pull := pj.Spec.Refs.Pulls[0]
+	org := pj.Spec.Refs.Org
+	repo := pj.Spec.Refs.Repo
+	if pcfg == nil {
```

Comment:
I think it might be better to require that someone pass the flag and load the config if they want to use this feature